### PR TITLE
jslegendre-themeengine: update livecheck

### PR DESCRIPTION
**Pull request by khipp**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/j/jslegendre-themeengine.rb [hunk 1]
```diff
@@ -9,10 +9,13 @@
 
   livecheck do
     url :url
-    strategy :git do |tags|
+    regex(/^v?(\d+(?:\.\d+)*)\((\d+)\)$/i)
```
Comment (samford) [inline Casks/j/jslegendre-themeengine.rb:12]:
```suggestion
    regex(/^v?(\d+(?:\.\d+)+)(?:\((\d+)\))?$/i)
```

The upstream repository also has some older tags like `0.0.4` (from before the fork), so it probably wouldn't hurt to make the parenthetical suffix optional.

There aren't any tags that use only one number, so we should use the default `+` form of the standard regex (i.e., it's less likely to match something unexpected in the future).

File: Casks/j/jslegendre-themeengine.rb [hunk 2]
```diff
@@ -9,10 +9,13 @@
...
+    strategy :git do |tags, regex|
```
Comment (samford) [inline Casks/j/jslegendre-themeengine.rb:13]:
```suggestion
    strategy :github_latest do |json, regex|
```

Upstream has marked all releases before the latest (v1.0.0(118)) as pre-release, so we should probably use `GithubLatest` here.

File: Casks/j/jslegendre-themeengine.rb [hunk 3]
```diff
@@ -9,10 +9,13 @@
...
       tags.filter_map do |tag|
-        match = tag.match(/^v(\d+(?:\.\d+)*)\((\d+)\)$/i)
-        "#{match[1]},#{match[2]}" if match
+        match = tag.match(regex)
+        next if match.blank?
+
+        "#{match[1]},#{match[2]}"
```
Comment (samford) [inline Casks/j/jslegendre-themeengine.rb:18]: Unfortunately GitHub (still) doesn't support suggestions that span deleted/replaced lines but this is what I would have suggested:

```ruby
      match = json["tag_name"]&.match(regex)
      next if match.blank?

      match[2] ? "#{match[1]},#{match[2]}" : match[1]
```

Comment (khipp) [reply Casks/j/jslegendre-themeengine.rb]: Makes sense, thank you!

**Review by samford — APPROVED** [review]

**Pull request merged**